Added flag to create DS9 region files

RaghavGirgaonkar opened the pull request:
Add flag ``` --reg``` to toggle creation of a ds9 region file if sources are present in search. Regions are circles and radius of the circle in arcsec can be supplied by the ``` --reg-radius``` flag, default value is 15. 

API use: ```vast_mw.vast_mw.create_regfile() ```

Added ```__init__.py``` files in ```vast_mw/``` and ```vast_mw/scripts``` and replaced ```setup.py``` with a ```pyproject.toml``` for ease of installation.

RaghavGirgaonkar committed:
Added option to create ds9 region file

RaghavGirgaonkar committed:
minor change to reg file flag and update README

RaghavGirgaonkar commented:
Correction: no ```pyproject.toml``` file included yet.

dlakaplan commented:
This looks good, but I'd suggest a few changes:
* allow the region files to include the source name (I don't think they did?)
* Put the statement about writing the region file either through the logger (maybe `INFO`) or with a comment (`#`) or otherwise to STDERR.  This is so you can pipe the output straight into another script if needed and it will only contain the actual results.

RaghavGirgaonkar committed:
added logger entry to reg file creation

RaghavGirgaonkar committed:
added pyproject.toml and update README

RaghavGirgaonkar commented:
- Added the logger ```INFO``` entry for creation of reg files.
- The reg file automatically includes the source name it uses the ```vast_mw.vast_mw.format_name()``` routine, so 
```
s = vast_mw.format_name(source)
src = s.replace(" ", "_")
 ``` 
 giving
 ```
 check_gaia -c "11h05m21.52536s,+43d31m34.9932s" -t "2023-12-21 21:07:40" --radius=45 --reg -vv --url
DEBUG   : Input time is '2023-12-21 21:07:40.000'
INFO    : For source at '11h05m21.53s, +43d31m35.0s' = '166.340d, +43.526d', found 2 Gaia matches within 45.0 arcsec
VAST J1105.3+4332       Gaia DR3 778947608243864320:  6.3 arcsec        https://gaia.ari.uni-heidelberg.de/singlesource.html#gaiadr3_id=778947608243864320
VAST J1105.3+4332       Gaia DR3 778947814402602752: 37.5 arcsec        https://gaia.ari.uni-heidelberg.de/singlesource.html#gaiadr3_id=778947814402602752
INFO    : Wrote ds9 region file VAST_J1105.3+4332_check_gaia.reg
```
Added this example to the README 

- Also added a ```pyproject.toml``` file

dlakaplan commented on pyproject.toml:
```diff
@@ -0,0 +1,53 @@
+version = "0.1.1"
+description = "Multi-wavelength searches for VAST sources"
+authors = [
+    { name = "David Kaplan", email = "kaplan@uwm.edu" }
```
Add yourself as an author.

dlakaplan commented:
Note that you could use https://astropy-regions.readthedocs.io/en/stable/, but don't have to

dlakaplan commented:
Should the test region files have source names?

RaghavGirgaonkar committed:
updated pyproject.toml and fixed logger issue in check_all.py

RaghavGirgaonkar commented:
- Updated the ```pyproject.toml```
- Yes, the test regions will have the source name, created updated reg files. 
- Fixed a minor logger issue in ```check_all.py``` which wasn't printing ```log.info``` messages and was only printing ```log.warning``` messages, it can now switch between the two depending on the results of the services.

RaghavGirgaonkar committed:
added text option to regions, improved readability of test.ipynb and added api + cli tests

RaghavGirgaonkar commented:
- Modified ```create_regfile()``` so that each source has its name as region text.
- Improved readability of ```tests.ipynb``` and added examples of cli and api usage
